extensions/amp-carousel/0.1/carousel-controls.js

```diff
@@ -74,8 +75,8 @@ export class CarouselControls {
    * @param {*} onInteraction
    */
   setupButtonInteraction(button, onInteraction) {
-    button.addEventListener('click', onInteraction);
-    button.addEventListener('keydown', (event) => {
+    /** @param {KeyboardEvent} event*/
+    function onKeyDown(event) {
```

Comment:
Instead of rewriting these (which is very very likely to miss a `this` or similar), I'd recommend adding a "fn name" param to `addSerializedEventListener`.